### DISCUSSION TITLE
(Internal) CPG Hackathon 2024-10-02

### DISCUSSION THREAD

**Discussion by maximiliankaul** [Fraunhofer-AISEC/cpg#1725] [General]
Please add your ideas below.

- I'd like to get an introduction an the Control Flow Sensitive DFG pass (and other "basic passes" without good documentation one should know about).
- Please tell me more about how the graph works. Why are we storing lists of edges. What's going on with all the funny annotations? Which ones should I know? What's the unwrap stuff? ...

**Comment by morbitzer** [comment]
+ Difference between the DeclarationState and the generalState (applies to all passes)

**Reply by oxisto** [reply]
> * Difference between the DeclarationState and the generalState (applies to all passes)

Isn’t this only used in the CFDFG?

**Reply by morbitzer** [reply]
Ok, maybe it only applies to the Alias and the (Alias)CFSDFGPass

**Comment by oxisto** [comment]
If there is time we could discuss about the possible "withChildren" styles for the node builder (e.g., with `it`, with implicit receiver, with function, with block, etc.). I can prepare some different options,

**Reply by maximiliankaul** [reply]
I would welcome this topic. It's time to finalize this stuff.

**Comment by morbitzer** [comment]
+ Which DFG/(Alias)CFSDFG Pass is supposed to draw which edges?

**Comment by maximiliankaul** [comment]
Assuming this is relevant for our current project: what's the current status of the value evaluators / what should be the status / short introduction on how they currently work.

**Comment by maximiliankaul** [comment]
Focus for Wednesday: passes
- Mermaid dependency graph
- Annotations: ExecuteBefore, ... @maximiliankaul 
- Types of passes: component pass, ... @oxisto 
- EOG @konradweiss 
- Type Resolving @oxisto 
- Symbol Resolver @oxisto 
- DFG / CFSDFG @KuechA

**Comment by konradweiss** [comment]
What is documented where:
General Information, principles and Examples: https://github.com/Fraunhofer-AISEC/cpg/wiki
Documentation for users, specifications: https://fraunhofer-aisec.github.io/cpg/
Comments in code

**Comment by maximiliankaul** [comment]
Next Hackathon: Quary API QueryTree
